json/src/main/java/org/teavm/flavour/json/emit/ClassInformationProvider.java

```diff
@@ -479,7 +479,7 @@ private void scanFields(ClassInformation information, ReflectClass<?> cls) {
             if (Modifier.isStatic(field.getModifiers())) {
                 continue;
             }
-            if (hasExplicitPropertyDeclaration(field) || information.getterVisibility.match(field.getModifiers())) {
```

Comment:
By the way, at least this part should be taken or annotation approach would not work either.

Reply:
Right. You can submit a separate PR so that I could accept it.

Reply:
BTW, it's a good idea to include a test together with this fix